File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Internal/Activation/NullModifiedObjectQuery.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.Internal.Activation;

namespace Db4objects.Db4o.Internal.Activation
{
	public class NullModifiedObjectQuery : IModifiedObjectQuery
	{
		public static readonly IModifiedObjectQuery Instance = new Db4objects.Db4o.Internal.Activation.NullModifiedObjectQuery
			();

		private NullModifiedObjectQuery()
		{
		}

		public virtual bool IsModified(object @ref)
		{
			return false;
		}
	}
}

File: db4oj/core/src/com/db4o/internal/activation/NullModifiedObjectQuery.java
package com.db4o.internal.activation;

public class NullModifiedObjectQuery implements ModifiedObjectQuery {
	
	public final static ModifiedObjectQuery INSTANCE = new NullModifiedObjectQuery();

	private NullModifiedObjectQuery() {
	}
	
	public boolean isModified(Object ref) {
		return false;
	}

}
